make separator configurable

pirxpilot opened the pull request:
it's still comma by default but you can pass `{ separator: ' ' }` or `{ separator: ':' }` as an option argument when creating `tags-input`

pirxpilot committed:
make separator configurable through `data-separator`

`data-separator` attribute on a wrapped input element can be used to
specify tags separator

We use `KeyboardEvent.key` to detect separator, if browser does not
support it (Safari) we fall back to KeyboardEvent.keyIdentifier

see:
http://caniuse.com/#search=KeyboardEvent.key

developit commented:
Thought: what about having it be an attribute on the input?

```html
<input type="tags" data-separator=":" value="foo:bar:baz">
```

pirxpilot commented:
Yup. Much better. Let me rework it.

pirxpilot commented:
Some fixes:
- implemented `data-separator`
- fixed char detection - Safari does not actually support neither `key` nor `char` on `keydown` events - we need to parse [keyIdentifier] to properly detect separator char

[keyIdentifier]: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyIdentifier

luckydonald commented:
How can one specify multible seperators? E.g. `","` and <kbd>enter</kbd>?

developit commented on src/tags-input.js:
```diff
@@ -104,8 +101,22 @@ export default function tagsInput(input) {
+			return String.fromCharCode(parseInt(event.keyIdentifier.slice(2), 16));
+		}
+		// other old/non-conforming browsers
+		return e.char;
```
hmm - I think we could just use `e.charCode` here, couldn't we?

developit commented on src/tags-input.js:
```diff
@@ -139,14 +150,15 @@ export default function tagsInput(input) {
 	base.input.addEventListener('keydown', e => {
 		let el = base.input,
 			key = e.keyCode || e.which,
+			char = charFromKeyboardEvent(e),
```
hmm - I think we could just use `String.fromCharCode(key)` here, couldn't we?

pirxpilot replied:
I wish. If only Safari behaved ;-)

developit replied:
but its using `keyCode`, not `charCode`. I thought that worked? 😥

pirxpilot replied:
Yeah - `keyCode` works in most browsers on `keypress` event. But on `keydown` a different story.
Mobile (iOS) Safari tends to have strange values in `keyCode` on `keydown` - for example `,` (comma) generates 188 which corresponds to '¼'. I am sure that there is some logic behind it, and I am sure that we are up for yet another _how to translate key events to key value standard_ at some point, but for now we are still in the waving hands and magic land.

developit commented:
@luckydonald by pasting

pirxpilot commented:
@luckydonald - `enter` and `tab` is _always_ a separator, as for others you can specify a string with multiple charactes - see: https://pirxpilot.github.io/tags-input/
Source for the demo is here: https://github.com/code42day/tags-input/blob/next/index.html

developit commented:
Ah now I see what was being asked haha.
Nice demo @pirxpilot

pirxpilot commented:
Detecting keys is tricky, not all documented properties are set on all events. For example `charCode` is not set in most browsers on `keyDown`.
see: https://dvcs.w3.org/hg/d4e/raw-file/tip/key-event-test.html

0xc00010ff commented:
🙏

tumpio commented:
@0xc00010ff check @pirxpilot 's fork 
https://github.com/pirxpilot/tags-input
